packages/device_info_plus/device_info_plus/lib/device_info_plus.dart

```diff
@@ -76,4 +79,25 @@ class DeviceInfoPlugin {
   /// Returns device information for Windows.
   Future<WindowsDeviceInfo> get windowsInfo async =>
       _cachedWindowsDeviceInfo ??= await _platform.windowsInfo()!;
+
+  /// Returns device information for the current platform.
+  Future<BaseDeviceInfo> get deviceInfo async {
+    if (kIsWeb) {
+      return webBrowserInfo;
+    } else {
+      if (Platform.isAndroid) {
+        return androidInfo;
+      } else if (Platform.isIOS) {
+        return iosInfo;
+      } else if (Platform.isLinux) {
+        return linuxInfo;
+      } else if (Platform.isMacOS) {
+        return macOsInfo;
+      } else if (Platform.isWindows) {
+        return windowsInfo;
+      }
+    }
+
+    throw UnsupportedError('Unsupported platform');
```

Comment:
The error message probably should tell the user which the current platform is. We could also tell the user to create an issue or PR in this repo for unsupported platforms.

Also, does this need tests?

Reply:
I'm not sure about the idea of more detailed explanation. Most `UnsupportedError` exceptions in `flutter/plugins` or this repo simply states that current platform is well not supported, without any further explanation. Thats contrasting with assertion messages throughout Flutter, which are very verbose, suggesting fixes, or further actions.